glean-core/ios/GleanTests/HttpPingUploaderTests.swift

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+import OHHTTPStubs
+@testable import Glean
+
+class HttpPingUploaderTests: XCTestCase {
+    var expectation: XCTestExpectation?
+    private let testPath = "/some/random/path/not/important"
+    private let testPing = "{ \"ping\": \"test\" }"
+    private let testConfig = Configuration(
+        userAgent: "Glean/Test 25.0.2",
+        connectionTimeout: 3050,
+        logPings: true,
+        pingTag: "Tag"
+    )
+
+    private func setupHttpResponseStub(statusCode: Int32 = 200) {
+        stub(condition: isHost("incoming.telemetry.mozilla.org")) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let str = String(data: body!, encoding: .utf8) ?? ""
+            print("STUB RECEIVED: \(str) \n")
+            // swiftlint:disable force_try
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            XCTAssertEqual(json?["ping"] as? String, "test")
+
+            return OHHTTPStubsResponse(
+                jsonObject: ["foo": "bar"],
+                statusCode: statusCode,
+                headers: ["Content-Type": "application/json"]
+            )
+        }
+    }
+
+    func test2XX() {
+        var testValue = false
+        setupHttpResponseStub(statusCode: 200)
+
+        expectation = expectation(description: "Completed upload")
+
+        let httpPingUploader = HttpPingUploader()
+        httpPingUploader.upload(path: testPath, data: testPing, config: testConfig) { success, _ in
+            testValue = success
+            self.expectation?.fulfill()
+        }
+        waitForExpectations(timeout: 10.0) { error in
```

Comment:
That's a rather long timeout. Is it necessary?

Reply:
looks like 2 seconds is plenty locally

Reply:
Since we're using timeout constants to please CI elsewhere in the Glean SDK (Android..), what about using the same value everywhere? I think we went for 5s or so